### PR TITLE
CNI helper package

### PR DESCRIPTION
**Pull request by ivan4th**
- [x] Add CNI / netlink deps
- [x] Escape veth pair creation
- [x] General bridge creation code
- [x] CNI interface info extraction / removal
- [x] Prepare tap interface
- [x] Container-side bridge: tap+calico_veth+dhcp_outer_veth
- [x] Emulator wrapper (enter ns, create dhcp ns, run dhcp server and the actual emulator)
- [x] Update build system to be able to put the wrapper in libvirt container
- [x] Update README.md

To be done in follow-up PRs:
- [ ] Introduce "user-facing" Makefile
- [ ] Check network connectivity in e2e
- [ ] Fix default route handling

---
This change is [Reviewable](https://reviewable.io/reviews/mirantis/virtlet/126)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,7 +2,7 @@ FROM ubuntu:16.04
 MAINTAINER Michal Rostecki <mrostecki@mirantis.com>
 LABEL Name="virtlet" Version="0.1"
 
-# TODO: dhcpcd5 is only needed for integration tests
+# TODO: dhcpcd5, ebtables and tcpdump are only needed for integration tests
```
Comment (jellonek) [inline Dockerfile]: Looks like `ebtables` are used not only in integration tests, but always in bridge setup.

File: pkg/nettools/nettools.go [hunk 2]
```diff
@@ -318,53 +301,27 @@ func SetupContainerSideNetwork(info *types.Result) (*ContainerNetwork, error) {
...
-		if err != nil {
-			return fmt.Errorf("failed to find dhcp side veth: %v", err)
-		}
-		if err = netlink.LinkSetUp(veth); err != nil {
-			return fmt.Errorf("failed to set %q up: %v", dhcpVethDhcpEndName, err)
-		}
-		if err = netlink.AddrAdd(veth, mustParseAddr(internalDhcpAddr)); err != nil {
-			return fmt.Errorf("failed to set address for dhcp side veth: %v", err)
+	// block DHCP traffic from/to CNI-provided link
+	for _, item := range []struct{ chain, opt string }{
```
Comment (jellonek) [inline pkg/nettools/nettools.go]: Additional spaces around struct elements.
Linter (probably `go vet`) will help to catch such things.

File: pkg/nettools/nettools_test.go [hunk 3]
```diff
@@ -54,9 +54,15 @@ var expectedExtractedLinkInfo = types.Result{
...
-		t.Fatalf("Error creating network namespace: %v", err)
+		t.Errorf("Error creating network namespace: %v", err)
+		return
 	}
 	defer func() {
+		// We use log.Panicf() instead of t.Fatalf() in these tests
+		// because ns.Do() uses separate goroutine
+		if r := recover(); r != nil {
+			t.Fatal(r)
+		}
```
Comment (jellonek) [inline pkg/nettools/nettools_test.go]: Good one 👍

**Comment by jellonek** [comment]
Reviewed 26 of 26 files at r1.
Review status: all files reviewed at latest revision, 19 unresolved discussions.

---

*[Dockerfile, line 5 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIbZqu8lo45aVzqz7r:-KYIbZqu3dAVMOGjK6RT:b-hihtxz) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/Dockerfile#L5)):*
> ```Dockerfile
> LABEL Name="virtlet" Version="0.1"
> 
> # TODO: dhcpcd5, ebtables and tcpdump are only needed for integration tests
> ```

It's not exactly the trus - ebtables is used not only in tests.

---

*[cmd/vmwrapper/vmwrapper.go, line 20 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIc7cW0NJYEljw_O87:-KYIc7cW0beolJBYux7V:b-djmpzq) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L20)):*
> ```Go
> 
> import (
> 	"encoding/json"
> ```

Again - you should use tabs instead of spaces.
Is your ide using https://golang.org/cmd/gofmt/ ?

---

*[cmd/vmwrapper/vmwrapper.go, line 50 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjWLeDzSrjoGOue31:-KYIjWLe4eJcPjsYPwSS:b-nv0htq) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L50)):*
> ```Go
> 
> 	if len(os.Args) < 2 {
> 		glog.Errorf("must specify emulator executable")
> ```

This should be started with capital letter.

---

*[cmd/vmwrapper/vmwrapper.go, line 58 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjXAT9VGimVg9qHx2:-KYIjXAU3xR3BgIny_Dk:bribihm) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L58)):*
> ```Go
> 	nsPath := os.Getenv(nsEnvVar)
> 	if nsPath == "" {
> 		glog.Errorf("must specify VIRTLET_NS")
> ```

As above.

---

*[cmd/vmwrapper/vmwrapper.go, line 64 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIj_Qe9e4DbphYej3l:-KYIj_QeBrSzC7rmlffJ:bribihm) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L64)):*
> ```Go
> 	vmNS, err := ns.GetNS(nsPath)
> 	if err != nil {
> 		glog.Errorf("failed to open network namespace at %q: %v", nsPath, err)
> ```

As above.

---

*[cmd/vmwrapper/vmwrapper.go, line 70 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIja902GgMgNBb1xbb:-KYIja90B15LPmX2ioJz:bribihm) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L70)):*
> ```Go
> 	peerHwAddrStr := os.Getenv(peerHwAddrEnvVar)
> 	if peerHwAddrStr == "" {
> 		glog.Errorf("must specify VIRTLET_HWADDR")
> ```

As above.

---

*[cmd/vmwrapper/vmwrapper.go, line 75 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjaso8ferZDIo5NHo:-KYIjasoB29mx8u6sSuA:bribihm) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L75)):*
> ```Go
> 	peerHwAddr, err := net.ParseMAC(peerHwAddrStr)
> 	if err != nil {
> 		glog.Errorf("bad hwaddr in VIRTLET_HWADDR")
> ```

As above.

---

*[cmd/vmwrapper/vmwrapper.go, line 82 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjLRn9890CIBAnKBO:-KYIjLRn3kGn7d4cLdwS:b-nv0htq) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L82)):*
> ```Go
> 	if cniConfig != "" {
> 		if err := json.Unmarshal([]byte(cniConfig), &info); err != nil {
> 			glog.Errorf("failed to unmarshal cni config: %v", err)
> ```

This should be started with capital letter.

---

*[cmd/vmwrapper/vmwrapper.go, line 82 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjbc-9PpWTKVBmvpz:-KYIjbc-5B9qpik9efxN:bribihm) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L82)):*
> ```Go
> 	if cniConfig != "" {
> 		if err := json.Unmarshal([]byte(cniConfig), &info); err != nil {
> 			glog.Errorf("failed to unmarshal cni config: %v", err)
> ```

As above.

---

*[cmd/vmwrapper/vmwrapper.go, line 117 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjePO99eIv5c5snU8:-KYIjePOCY9YQzRehOTx:bribihm) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L117)):*
> ```Go
> 		case dhcpErr := <-errCh:
> 			if dhcpErr == nil {
> 				glog.Errorf("dhcp server self-exited with nil error")
> ```

As above.

---

*[cmd/vmwrapper/vmwrapper.go, line 119 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjf4ZA3G2XCG9_VpH:-KYIjf4Z-OdA6HSFZ35s:bribihm) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L119)):*
> ```Go
> 				glog.Errorf("dhcp server self-exited with nil error")
> 			} else {
> 				glog.Errorf("dhcp server failed: %v", dhcpErr)
> ```

As above.

---

*[cmd/vmwrapper/vmwrapper.go, line 130 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjiVlANQQ1lyGLWK1:-KYIjiVl0H-NXKUaCP_w:b-pgg40v) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L130)):*
> ```Go
> 		return nil
> 	}); err != nil {
> 		glog.Error(err)
> ```

Returned errors are lowercased - they should be appended there to something starting with upper case.

---

*[contrib/docker-compose-test/docker-compose.yml, line 16 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjwqR09SLkwMGLLdi:-KYIjwqR3iBhoLNLIRJL:b-vtp5au) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/contrib/docker-compose-test/docker-compose.yml#L16)):*
> ```YAML
>   virtlet_test:
>     build: ../..
>     # 'privileged' is necessary for network namespaces
> ```

After merging with master this will be misleading, because it's also needed by CNI part of virtlet.

---

*[pkg/dhcp/config.go, line 27 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIk7my1Ab-z0A-apVl:-KYIk7myBgR0LPEgGF-D:by2ajtj) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/pkg/dhcp/config.go#L27)):*
> ```Go
> 
> type Config struct {
> 	// TODO :don't use CNI result type here
> ```

Please replace space-coma with coma-space.

---

*[pkg/dhcp/server.go, line 23 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIkN6GBztyoZrhpmyA:-KYIkN6G4DHi2jbDR8p-:b-fhowcv) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/pkg/dhcp/server.go#L23)):*
> ```Go
> 
> import (
> 	"bytes"
> ```

Again, tabs vs spaces.

---

*[pkg/dhcp/server.go, line 162 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIkqSk0dyI7mo9wVul:-KYIkqSkCOhIt7bzvNzZ:b-q8k90h) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/pkg/dhcp/server.go#L162)):*
> ```Go
> 	}
> 
> 	p.YourAddr = s.config.CNIResult.IP4.IP.IP
> ```

There is no verification if `s.config.CNIResult.IP4` is nil (can be if CNI ipam will be configured to server only ipv6 addresses). IMO this should be covered in time of wrapper start, after CNIResult unmarshal (so we could break then rest of execution, even before starting an emulator).

---

*[pkg/nettools/nettools.go, line 37 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIlw1O1ughlRF6aRRp:-KYIlw1O0X4wX1FKJ3hg:bfcy6z7) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/pkg/nettools/nettools.go#L37)):*
> ```Go
> 
> import (
> 	"errors"
> ```

Again. Spaces vs tabs.

---

*[pkg/nettools/nettools.go, line 259 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYImSMW4oifkCNRwMFB:-KYImSMW7LeNaJbAVPFt:bl11ny4) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/pkg/nettools/nettools.go#L259)):*
> ```Go
> 	r, err := netlink.ParseAddr(addr)
> 	if err != nil {
> 		log.Panicf("failed to parse address %q: %v", addr, err)
> ```

Error message should be capitalized.

---

*[tests/integration/vm_network_test.go, line 20 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIbqABCjPjdqnq-WRi:-KYIbqAB-13qiaHs6AiE:b-q3aimp) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/tests/integration/vm_network_test.go#L20)):*
> ```Go
> 
> import (
> 	"bytes"
> ```

Spaces instead of tabs?
Looks like you have misconfigured IDE.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/126)*

**Comment by ivan4th** [comment]
Review status: all files reviewed at latest revision, 19 unresolved discussions.

---

*[Dockerfile, line 5 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIbZqu8lo45aVzqz7r:-KYK16hu9fcQiunJhxHU:bgad0ts) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/Dockerfile#L5)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

It's not exactly the trus - ebtables is used not only in tests.
</blockquote></details>

I meant that it's only needed for tests in this image. libvirt image is what actually uses it (because vmwrapper runs there)

---

*[cmd/vmwrapper/vmwrapper.go, line 20 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIc7cW0NJYEljw_O87:-KYK1SC7D31YQmNasZrX:bfhl664) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L20)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

Again - you should use tabs instead of spaces.
Is your ide using https://golang.org/cmd/gofmt/ ?
</blockquote></details>

(illusion caused by reviewable, these are tabs)

---

*[pkg/dhcp/server.go, line 23 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIkN6GBztyoZrhpmyA:-KYK6ya_3c9UK5q7ridj:b-6anxia) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/pkg/dhcp/server.go#L23)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

Again, tabs vs spaces.
</blockquote></details>

reviewable's illusion, see above

---

*[pkg/dhcp/server.go, line 162 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIkqSk0dyI7mo9wVul:-KYK7K3aB33YqeEEb16V:brks5dr) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/pkg/dhcp/server.go#L162)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

There is no verification if `s.config.CNIResult.IP4` is nil (can be if CNI ipam will be configured to server only ipv6 addresses). IMO this should be covered in time of wrapper start, after CNIResult unmarshal (so we could break then rest of execution, even before starting an emulator).
</blockquote></details>

Thanks, fixing this.

---

*[pkg/nettools/nettools.go, line 37 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIlw1O1ughlRF6aRRp:-KYK7bvX4CPWUQUTG3C6:b-hf862y) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/pkg/nettools/nettools.go#L37)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

Again. Spaces vs tabs.
</blockquote></details>

illusion, see above

---

*[tests/integration/vm_network_test.go, line 20 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIbqABCjPjdqnq-WRi:-KYK7l1O3Z6J_X0-M4qE:bngqtzo) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/tests/integration/vm_network_test.go#L20)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

Spaces instead of tabs?
Looks like you have misconfigured IDE.
</blockquote></details>

again, reviewable's illusion

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/126)*

File: pkg/nettools/nettools.go (hunk 2 above)
Comment (ivan4th) [inline pkg/nettools/nettools.go]: The spaces are forced by go-fmt.

**Comment by jellonek** [comment]
Reviewed 9 of 19 files at r2, 26 of 27 files at r3.
Review status: all files reviewed at latest revision, 15 unresolved discussions, some commit checks failed.

---

*[README.md, line 21 at r3](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYYy0aJb7iSnQHacOo6:-KYYy0aJb7iSnQHacOo7:b-pacy30) ([raw file](https://github.com/mirantis/virtlet/blob/fcb7ce5d3fc6012a9773a72c590feeb521ce4198/README.md#L21)):*
> ```Markdown
> ### Networking support
> 
> For now the only supported configuration is the use of [CNI plugins](https://github.com/containernetworking/cni) which are required to be installed on the host.
> ```

Binaries are added into virtlet image so it's not exactly the true.

---

*[cmd/vmwrapper/vmwrapper.go, line 73 at r3](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYYyNFScleIwGyTJZeE:-KYYyNFScleIwGyTJZeF:btesh3e) ([raw file](https://github.com/mirantis/virtlet/blob/fcb7ce5d3fc6012a9773a72c590feeb521ce4198/cmd/vmwrapper/vmwrapper.go#L73)):*
> ```Go
> 
> 	// FIXME
> 	os.Args = []string{os.Args[0], "-v=3", "-alsologtostderr=true"}
> ```

This is probably debug option, which should be removed before merge.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/126)*

**Comment by ivan4th** [comment]
Review status: all files reviewed at latest revision, 15 unresolved discussions, some commit checks failed.

---

*[cmd/vmwrapper/vmwrapper.go, line 50 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjWLeDzSrjoGOue31:-KYZ3N7GHXTE_kUz42DK:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L50)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

This should be started with capital letter.
</blockquote></details>

Done.

---

*[cmd/vmwrapper/vmwrapper.go, line 58 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjXAT9VGimVg9qHx2:-KYZ3OgC1Xrz1HGRY5kB:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L58)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

As above.
</blockquote></details>

Done.

---

*[cmd/vmwrapper/vmwrapper.go, line 64 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIj_Qe9e4DbphYej3l:-KYZ3QO8vdNMGZs366td:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L64)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

As above.
</blockquote></details>

Done.

---

*[cmd/vmwrapper/vmwrapper.go, line 70 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIja902GgMgNBb1xbb:-KYZ3Qvq3FftpzsvZy5V:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L70)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

As above.
</blockquote></details>

Done.

---

*[cmd/vmwrapper/vmwrapper.go, line 75 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjaso8ferZDIo5NHo:-KYZ3RvWzGkRAKiNl5Ox:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L75)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

As above.
</blockquote></details>

Done.

---

*[cmd/vmwrapper/vmwrapper.go, line 82 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjLRn9890CIBAnKBO:-KYZ3SWPJ8A616C0YCFU:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L82)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

This should be started with capital letter.
</blockquote></details>

Done.

---

*[cmd/vmwrapper/vmwrapper.go, line 82 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjbc-9PpWTKVBmvpz:-KYZ3TEbyUCvrp_4rxZY:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L82)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

As above.
</blockquote></details>

Done.

---

*[cmd/vmwrapper/vmwrapper.go, line 117 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjePO99eIv5c5snU8:-KYZ3U2JfHo6BA9H4353:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L117)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

As above.
</blockquote></details>

Done.

---

*[cmd/vmwrapper/vmwrapper.go, line 119 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjf4ZA3G2XCG9_VpH:-KYZ3US0dsuk_GKkHrvV:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L119)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

As above.
</blockquote></details>

Done.

---

*[cmd/vmwrapper/vmwrapper.go, line 130 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjiVlANQQ1lyGLWK1:-KYZ3VaboqE4T_MiQLPc:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/cmd/vmwrapper/vmwrapper.go#L130)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

Returned errors are lowercased - they should be appended there to something starting with upper case.
</blockquote></details>

Done.

---

*[cmd/vmwrapper/vmwrapper.go, line 73 at r3](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYYyNFScleIwGyTJZeE:-KYZ2qfVbSYw01scgcR0:bprw2i5) ([raw file](https://github.com/mirantis/virtlet/blob/fcb7ce5d3fc6012a9773a72c590feeb521ce4198/cmd/vmwrapper/vmwrapper.go#L73)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

This is probably debug option, which should be removed before merge.
</blockquote></details>

Actually the problem is that without logging it's quite hard to diagnose vmwrapper problems. The proper way to solve this is not forcing verbosity to zero but rather adding an extra env var to pass in verbosity value and using VIRTLET_VERBOSE on virtlet side to provide the value for it in <qemu:command-line>. But perhaps this should be done in separate PR? Moreover, this can be simplified if we decide to drop libvirt

---

*[contrib/docker-compose-test/docker-compose.yml, line 16 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIjwqR09SLkwMGLLdi:-KYZ3Y0n0V2-0YRN5gba:b-rxotv9) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/contrib/docker-compose-test/docker-compose.yml#L16)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

After merging with master this will be misleading, because it's also needed by CNI part of virtlet.
</blockquote></details>

Name updated

---

*[pkg/dhcp/config.go, line 27 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYIk7my1Ab-z0A-apVl:-KYZ3_YXRAzRtCrOztxQ:bb6un1j) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/pkg/dhcp/config.go#L27)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

Please replace space-coma with coma-space.
</blockquote></details>

Done

---

*[pkg/nettools/nettools.go, line 259 at r1](https://reviewable.io:443/reviews/mirantis/virtlet/126#-KYImSMW4oifkCNRwMFB:-KYZ3b1KEe9iK4t_bMoW:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/dbdb30bf14b81526d9c2b3e357eb73b23c676841/pkg/nettools/nettools.go#L259)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

Error message should be capitalized.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/126)*

**Comment by jellonek** [comment]
:lgtm:

---

Reviewed 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/126#-:-KYZFZVZlYQmaOOqJfUm:bnfp4nl)*

**Pull request merged**